s4e-backend/src/test/java/pl/cyfronet/s4e/controller/InvitationControllerTest.java

```diff
@@ -0,0 +1,163 @@
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static pl.cyfronet.s4e.Constants.API_PREFIX_V1;
+import static pl.cyfronet.s4e.TestJwtUtil.jwtBearerToken;
+
+@BasicTest
+@Slf4j
+@AutoConfigureMockMvc
+public class InvitationControllerTest {
+    @Autowired
+    private InstitutionRepository institutionRepository;
+
+    @Autowired
+    private AppUserRepository appUserRepository;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private TestDbHelper testDbHelper;
+
+    @Autowired
+    private SlugService slugService;
+
+    @Autowired
+    private GroupRepository groupRepository;
+
+    @Autowired
+    private UserRoleRepository userRoleRepository;
+
+    @Autowired
+    private InvitationRepository invitationRepository;
+
+    private AppUser manager;
+    private AppUser user;
+    private Institution institution;
+
+    @BeforeEach
+    public void beforeEach() {
+        reset();
+
+        institution = institutionRepository.save(InvitationHelper.institutionBuilder().build());
+        Group group = groupRepository
+                .save(InvitationHelper.defaultGroupBuilder(institution).build());
+
+        manager = appUserRepository.save(InvitationHelper.userBuilder().build());
+        addRoles(manager, group, new AppRole[]{AppRole.INST_ADMIN, AppRole.INST_MANAGER, AppRole.GROUP_MEMBER});
+
+        user = appUserRepository.save(InvitationHelper.userBuilder().build());
+    }
+
+    @AfterEach
+    public void afterEach() {
+        reset();
+    }
+
+    @Test
+    public void shouldCreate() throws Exception {
+        val email = "test@mail.pl";
+        CreateInvitationRequest request = CreateInvitationRequest
+                .builder()
+                .email(email)
+                .build();
+        val URL = API_PREFIX_V1 + "/institutions/{institution}/invitations";
+        mockMvc.perform(post(URL, institution.getSlug())
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .with(jwtBearerToken(manager, objectMapper))
+                    .content(objectMapper.writeValueAsBytes(request))
+        )
+        .andExpect(status().isOk());
+
+        val dbInvitation = invitationRepository
+                .findByEmailAndInstitutionSlug(
+                        email,
+                        institution.getSlug(),
+                        Invitation.class
+                );
+        assertThat(dbInvitation, isPresent());
+
+        val invitation = dbInvitation
+                .orElseThrow(() -> new NotFoundException());
```

Comment:
`val invitation = dbInvitation.get()` would suffice as you check for presence above.